lib/pbench/server/indexing_tarballs.py

```diff
@@ -321,6 +345,18 @@ def sigquit_handler(*args):
                                 retries,
                             )
                             tb_res = 1 if failures > 0 else 0
+                        finally:
+                            if dataset:
+                                try:
+                                    dataset.advance(
+                                        States.INDEXED
+                                        if tb_res == 0
+                                        else States.QUARANTINED
+                                    )
+                                    Metadata.remove(dataset, Metadata.REINDEX)
```

Comment:
If the key is not found, this is a no-op?

Reply:
Right; `Metadata.REINDEX` will be set only if `pbench-reindex` provoked the indexing. We remove that key to clean it up, but if it wasn't there that's fine.

In the long run, I expect we'll be looking "for datasets that haven't been indexed and for which reindex has been requested" ... for now, I'm trying to add the persistent state, but our components are still exclusively using the old file link "state" subdirectories to make their decisions.

Reply:
I added a unit test case to prove that `Metadata.remove` on an unset key isn't an error.